src/passes/DeadArgumentElimination.cpp

```diff
@@ -299,12 +299,12 @@ struct DAE : public Pass {
               value = c->value;
             } else if (value != c->value) {
               // Not identical, give up
-              value.type = Type::none;
+              value = Literal();
```

Comment:
This is one of the occasions where I suspected that changing the type without disposing the payload first might be problematic, and why I made `Literal::type` immutable. As a result, this now utilizes the assignment operator to void the value.